### PR TITLE
DDF-5703 Extract core features from apps

### PR DESCRIPTION
**Pull request by pklinef**
#### What does this PR do?
Extracts the core catalog features from `-app` features.  Creates a new `ddf-catalog` distribution based on these features that is 300mb in size.

#### Ask 2 committers to review/merge the PR and tag them here.

@coyotesqrl
@rzwiefel

#### How should this be tested?

- Verify DDF distribution still works using the standard profile
- Verify `distribution/ddf-catalog` distribution by running `feature:install profile-catalog` at the command line
- Ingest data using command line and/or REST endpoint
- Query using command line and/or OpenSearch endpoint

#### What are the relevant tickets?
Fixes: #5703

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by pklinef** [comment]
build now

File: catalog/catalog-app/src/main/resources/features.xml [hunk 1]
```diff
@@ -99,26 +101,203 @@
...
+             description="CSV Query Response Transformer and Common Library.">
+        <bundle>mvn:ddf.catalog.transformer/csv-queryresponse-transformer/${project.version}</bundle>
+    </feature>
+
+    <feature name="catalog-transformer-geoformatter" version="${project.version}"
+             description="Geo library to help with conversion of geometry objects into various formats such GeoJson, GeoRSS, etc.">
+        <feature>abdera</feature>
+        <bundle>mvn:ddf.catalog.common/geo-formatter/${project.version}</bundle>
+        <bundle>mvn:org.apache.abdera/abdera-extensions-geo/${abdera.version}</bundle>
+        <bundle>mvn:com.googlecode.json-simple/json-simple/1.1.1</bundle>
```
Comment (coyotesqrl) [inline catalog/catalog-app/src/main/resources/features.xml]: ✏️ Should be able to use the property `json-simple.version`.

File: catalog/catalog-app/src/main/resources/features.xml [hunk 2]
```diff
@@ -99,26 +101,203 @@
...
+    </feature>
+
+    <feature name="abdera" version="${project.version}"
+             description="Apache Abdera">
+        <bundle>wrap:mvn:org.apache.james/apache-mime4j-core/0.7.2</bundle>
+        <bundle>mvn:commons-codec/commons-codec/1.4</bundle>
+        <!-- This bundle <bundle>mvn:org.apache.ws.commons.axiom/axiom-impl/1.2.10</bundle>
+        does not export any classes. Must use the servicemix version. -->
+        <bundle>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.axiom-impl/1.2.12-2</bundle>
+        <bundle>mvn:org.apache.ws.commons.axiom/axiom-api/1.2.14</bundle>
```
Comment (coyotesqrl) [inline catalog/catalog-app/src/main/resources/features.xml]: ✏️ Should be able to use the property `axiom.version` (defined in `/catalog.pom.xml`).

File: catalog/catalog-app/src/main/resources/features.xml [hunk 3]
```diff
@@ -99,26 +101,203 @@
...
+        <feature>cxf-rs-security-sso-saml</feature>
+        <feature>catalog-transformer-geoformatter</feature>
+        <bundle>mvn:org.apache.abdera/abdera-server/${abdera.version}</bundle>
+        <bundle>mvn:ddf.catalog.opensearch/catalog-opensearch-source/${project.version}</bundle>
+    </feature>
+
+    <feature name="abdera" version="${project.version}"
+             description="Apache Abdera">
+        <bundle>wrap:mvn:org.apache.james/apache-mime4j-core/0.7.2</bundle>
+        <bundle>mvn:commons-codec/commons-codec/1.4</bundle>
```
Comment (coyotesqrl) [inline catalog/catalog-app/src/main/resources/features.xml]: ❓ Can abdera use the `1.13` version used elsewhere, specified with the `commons-codec.version` property?

File: catalog/catalog-app/src/main/resources/features.xml [hunk 4]
```diff
@@ -99,26 +101,203 @@
...
         <bundle>mvn:ddf.catalog.core/catalog-core-metacardgroomerplugin/${project.version}</bundle>
     </feature>
 
     <feature name="catalog-core-content" version="${project.version}"
              description="Core features for content">
         <bundle>mvn:ddf.catalog.core/catalog-core-contentresourcereader/${project.version}</bundle>
         <bundle>mvn:org.codice.ddf/checksum/${project.version}</bundle>
+    </feature>
+
+    <feature name="catalog-rest" version="${project.version}">
```
Comment (ryeats) [inline catalog/catalog-app/src/main/resources/features.xml]: a few of these features are missing some dependencies in order to be standalone

Comment (pklinef) [reply catalog/catalog-app/src/main/resources/features.xml]: Features in the root features folder have been fixed to include all of their dependencies.  Catalog and spatial have not been fixed yet.

**Comment by pklinef** [comment]
build now

**Comment by pklinef** [comment]
build now

**Pull request merged**